src/redux/reducers/upload-reducer.js

```diff
@@ -77,6 +95,20 @@ const uploadReducer = (state = initState, action) => {
         history: h
       };
 
+    case uploadActions.INITIALIZE_POLLING_INDEXES:
+      const { frontIdx, backIdx, dataMapLength } = action.payload;
+      const indexes = {
+        ...state.indexes,
+        endingIdx: dataMapLength - 1,
+        frontIdx,
+        backIdx
+      };
+      return {
+        ...state,
+        indexes: indexes,
```

Comment:
just `indexes` works too, it's shorthand for `indexes: indexes`

Reply:
or if you want to shrink it even more, just take out the variable assignment altogether

```
+      indexes: {
+        ...state.indexes,
+        endingIdx: dataMapLength - 1,
+        frontIdx,
+        backIdx
+      };
```